src/LazyLieManifolds.jl

```diff
@@ -1,18 +1,86 @@
 
-export se, SE
+##==============================================================================
+## Abstract Lie Groups and Manifolds 
+##==============================================================================
+
+abstract type AbstractLieGroup end
+abstract type AbstractLieAlgebra end
```

Comment:
> If these new abstracts are just to reduce the function definitions, I think it would be better to write out the functions instead -- i.e. write against <: SE or <: se rather than new abstracts?

Yes, it is just to reduce function definitions. 

`AbstractLieGroup`  is not the correct name as it's a combination of a point and a manifold. 
Maybe something like `AbstractGroupManifoldPoint` and `AbstractGroupAlgebraVector` is more accurate.

That is this wrapper exists, lets continue in JuliaRobotics/RoME.jl#450